### PR TITLE
Add HassGetDate and HassGetTime

### PR DESCRIPTION
**Pull request by synesthesiam**
Add English sentences for two new intents (not yet supported in HA):

* `HassGetCurrentDate`
    * Receives a `date` slot value with a `datetime` object
    * Responds with the current date
* `HassGetCurrentTime`
    * Receives a `time` slot value with a `datetime` object
    * Responds with the current time

## Summary by CodeRabbit

- **New Features**
  - Introduced new intents `HassGetCurrentDate` and `HassGetCurrentTime` to retrieve the current date and time in English.
  - Added corresponding response templates for formatting and delivering date and time information.
  - Users can now query the current date and time through natural language commands.

- **Tests**
  - Added test cases to ensure accurate responses for the new date and time intents.

### CODE REVIEW

File: intents.yaml [hunk 1]
```diff
@@ -180,6 +180,16 @@ HassSetPosition:
...
 
+HassGetDate:
+  supported: true
+  domain: homeassistant
+  description: "Gets the current date"
+
+HassGetTime:
+  supported: true
+  domain: homeassistant
+  description: "Gets the current time"
```
Comment (frenck) [inline intents.yaml:191]: Considering Home Assistant also has date & time entities, these names seem very unfortunate for the future.

Wouldn't it be better to use `HassGetCurrentDate` and `HassGetCurrentTime` instead?

**Review by frenck — CHANGES_REQUESTED** [review]

File: intents.yaml (hunk 1 above)
Comment (synesthesiam) [reply intents.yaml:191]: That's a good point. I'll change them :+1:

Comment (synesthesiam) [reply intents.yaml]: Updated

**Review by frenck — APPROVED** [review]
Nice! 🎉 

../Frenck

**Pull request merged**